### ISSUE TITLE
Configuration validation accepts duplicate extensions in same naming rule array

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#127]
## Bug Description

A naming rule with duplicate extensions in the same array passes validation without any warning. For example, `extensions = ["ts", "tsx", "ts"]` where `"ts"` appears twice.

## Severity

**Low** - Likely harmless but indicates a configuration mistake that should be caught.

## Steps to Reproduce

1. Create a `check.toml` with duplicate extensions in the same rule:
```toml
[code.naming]
enabled = true

[[code.naming.rules]]
extensions = ["ts", "tsx", "ts"]  # "ts" appears twice
file_case = "kebab-case"
folder_case = "kebab-case"
```

2. Run `cm validate config`

## Expected Behavior

Validation should either:
- Warn about duplicate extensions within the same rule, OR
- Automatically deduplicate them

## Actual Behavior

```
✓ Valid
```

No warning is shown about the duplicate extension.

## Environment

- check-my-toolkit version: 1.3.0

## Suggested Fix

Add validation to detect and warn about duplicate extensions within the same naming rule array.
